indexing-service/src/main/java/org/apache/druid/indexing/overlord/hrtr/HttpRemoteTaskRunner.java

```diff
@@ -463,6 +463,32 @@ public void nodeViewInitialized()
         }
     );
 
+    DruidNodeDiscovery druidNodeDiscoveryIndexer = druidNodeDiscoveryProvider.getForNodeType(NodeType.INDEXER);
```

Comment:
Instead of repeated code you can change line 441 to

```
DruidNodeDiscovery druidNodeDiscovery = druidNodeDiscoveryProvider.getForService(WorkerNodeService.DISCOVERY_SERVICE_KEY);
```
```

Reply:
I changed this to use the suggested code.

It looks like `WorkerNodeService.DISCOVERY_SERVICE_KEY` was incorrectly mapped to `NodeType.PEON` in `DruidNodeDiscoveryProvider` previously, I also changed that to the MM and indexer types.